internal/contour/handler.go

```diff
@@ -164,11 +181,39 @@ func (e *EventHandler) Start(ctx context.Context) error {
 				// not to process it.
 				e.incSequence()
 			}
+
+			// We're done processing this event
+			if updateOpAdd, ok := op.(opAdd); ok {
+				if updateOpAdd.isInInitialList {
+					e.syncTracker.Finished()
+				}
+			}
 		case <-pending:
+			// Ensure informer caches are synced.
+			// Schedule a retry for dag rebuild if cache is not synced yet.
+			// Note that we can't block and wait for the cache sync as it depends on progress of this loop.
+			if !e.syncTracker.HasSynced() {
+				e.Info("skipping dag rebuild as cache is not synced")
+				timer.Reset(e.holdoffDelay)
+				break
+			}
+
 			e.WithField("last_update", time.Since(lastDAGRebuild)).WithField("outstanding", reset()).Info("performing delayed update")
-			e.rebuildDAG()
-			e.incSequence()
+
+			// Build a new DAG and sends it to the Observer.
+			latestDAG := e.builder.Build()
 			lastDAGRebuild = time.Now()
```

Comment:
Sorry, but can I still ask one more change? 😅 

Could we move `lastDAGRebuild = time.Now()` to last line after `e.incSequence()` to keep [the old way](https://github.com/projectcontour/contour/blob/39e86eefa4b0f298691d426d9968703949055f21/internal/contour/handler.go#L168-L171) how the time was measurement, that is, it included also the status update (even though the name says "DAG rebuild").  

I think it would be good to keep that included in time spent, since considerable time can go into status update when the go channel happens to block.

Reply:
Sure 😀. Done.